File: flowerwreath/flowerwreath_test.go
package flowerwreath_test

import (
	"testing"

	"github.com/hsmtkk/fuzzy-giggle/flowerwreath"
	"github.com/stretchr/testify/assert"
)

func Test1(t *testing.T) {
	got := flowerwreath.Solve(flowerwreath.Problem{"baaab", "aabba"})
	assert.True(t, got)
}

func Test2(t *testing.T) {
	got := flowerwreath.Solve(flowerwreath.Problem{"baaab", "abaab"})
	assert.False(t, got)
}

func TestRotate(t *testing.T) {
	got := flowerwreath.Rotate("baaab")
	want := map[string]bool{}
	for _, s := range []string{"baaab", "aaabb", "aabba", "abbaa", "bbaaa"} {
		want[s] = true
	}
	assert.Equal(t, want, got)
}

File: flowerwreath/flowerwreath_test.py
import unittest

import flowerwreath


class TestFlowerWreath(unittest.TestCase):
    def test1(self):
        problem = flowerwreath.Problem("baaab", "aabba")
        self.assertTrue(flowerwreath.solve(problem))

    def test2(self):
        problem = flowerwreath.Problem("baaab", "abaab")
        self.assertFalse(flowerwreath.solve(problem))


class TestRotate(unittest.TestCase):
    def test1(self):
        got = flowerwreath.rotate("baaab")
        want = {"baaab", "aaabb", "aabba", "abbaa", "bbaaa"}
        self.assertEqual(want, got)


if __name__ == "__main__":
    unittest.main()
